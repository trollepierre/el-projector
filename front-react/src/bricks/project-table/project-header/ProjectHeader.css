.project-header tr {
    display: grid;
    justify-content: space-around;
    grid-template-columns: repeat(10, 1fr [col-start]);
    height: 50px;
}

