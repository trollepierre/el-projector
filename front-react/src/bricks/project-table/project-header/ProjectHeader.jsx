import React from 'react';
import Button from '../../../components/button/Button';
import {
  sortByBirthday,
  sortByFirstName,
  sortByLastDate,
  sortByLastName,
  sortByNext,
  sortByOrigin,
  sortByPoints,
} from '../services/tasks-sorter';
// eslint-disable-next-line no-unused-vars
import styles from './ProjectHeader.css';

const ProjectHeader = () => (
  <thead className="project-header">
  <tr>
    <th className="id">
      <Button
        background-color="cyan"
        color="black'"
        text="id"
        onClick={sortByNext}
      />
    </th>
    <th className="name">
      <Button
        background-color="'cyan'"
        color="'black'"
        text="Name"
        onClick={sortByFirstName}
      />
    </th>
    <th className="description">
      <Button
        background-color="'cyan'"
        color="'black'"
        text="description"
        onClick={sortByLastName}
      />
    </th>
    <th className="points">
      <Button
        background-color="'cyan'"
        color="'black'"
        text="Points"
        onClick={sortByPoints}
      />
    </th>
    <th className="created-date">
      <Button
        background-color="'cyan'"
        color="'black'"
        text="created Date"
        onClick={sortByPoints}
      />
    </th>
    <th className="last-date">
      <Button
        background-color="'cyan'"
        color="'black'"
        text="LastDate"
        onClick={sortByLastDate}
      />
    </th>
    <th className="impact">
      Impact
    </th>
    <th className="birth-date">
      <Button
        background-color="'cyan'"
        color="'black'"
        text="BirthDate"
        onClick={sortByBirthday}
      />
    </th>
    <th className="origin">
      <Button
        background-color="'cyan'"
        color="'black'"
        text="Origin"
        onClick={sortByOrigin}
      />
    </th>
    <th className="action">
      Action
    </th>
  </tr>
  </thead>
);

export default ProjectHeader;
