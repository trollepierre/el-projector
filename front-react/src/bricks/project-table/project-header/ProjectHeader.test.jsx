import React from 'react';
import { shallow } from 'enzyme'
import ProjectHeader from './ProjectHeader';

describe('ProjectHeader', () => {
  describe('template', () => {
    it('should match snapshot', () => {
      // When
      const wrapper = shallow(<ProjectHeader/>)

      // Then
      expect(wrapper).toMatchSnapshot()
    });
  });
});
