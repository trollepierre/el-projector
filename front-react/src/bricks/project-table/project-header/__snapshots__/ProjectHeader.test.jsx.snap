// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectHeader template should match snapshot 1`] = `
<thead
  className="project-header"
>
  <tr>
    <th
      className="id"
    >
      <ProjectTable
        background-color="cyan"
        color="black'"
        onClick={[Function]}
        text="id"
      />
    </th>
    <th
      className="name"
    >
      <ProjectTable
        background-color="'cyan'"
        color="'black'"
        onClick={[Function]}
        text="Name"
      />
    </th>
    <th
      className="description"
    >
      <ProjectTable
        background-color="'cyan'"
        color="'black'"
        onClick={[Function]}
        text="description"
      />
    </th>
    <th
      className="points"
    >
      <ProjectTable
        background-color="'cyan'"
        color="'black'"
        onClick={[Function]}
        text="Points"
      />
    </th>
    <th
      className="created-date"
    >
      <ProjectTable
        background-color="'cyan'"
        color="'black'"
        onClick={[Function]}
        text="created Date"
      />
    </th>
    <th
      className="last-date"
    >
      <ProjectTable
        background-color="'cyan'"
        color="'black'"
        onClick={[Function]}
        text="LastDate"
      />
    </th>
    <th
      className="impact"
    >
      Impact
    </th>
    <th
      className="birth-date"
    >
      <ProjectTable
        background-color="'cyan'"
        color="'black'"
        onClick={[Function]}
        text="BirthDate"
      />
    </th>
    <th
      className="origin"
    >
      <ProjectTable
        background-color="'cyan'"
        color="'black'"
        onClick={[Function]}
        text="Origin"
      />
    </th>
    <th
      className="action"
    >
      Action
    </th>
  </tr>
</thead>
`;
